providers/default.rb

```diff
@@ -23,41 +23,76 @@ def nssm_exe
   "#{node['nssm']['install_location']}\\nssm.exe"
 end
 
-action :install do
-  if platform?('windows')
-    install_nssm
-    service_installed = service_installed?(new_resource.servicename)
+action :install_if_missing do
+  unless platform?('windows')
```

Comment:
I did have issues with linux platforms failing on windows specific stuff when windows resources were not inside the if block.  This should be tested against a linux platform with chef 12.6 to make sure this is not an issue on that client.